gcp-vertex-ai-mojo-scorer/src/main/java/ai/h2o/mojos/deploy/gcp/vertex/ai/controller/ModelsApiController.java

```diff
@@ -74,8 +86,45 @@ public static ScoreRequest getRestScoreRequest(
     request.setIdField(gcpRequest.getParameters().getIdField());
     request.setFields(gcpRequest.getParameters().getFields());
     
+    // Check if a pre-processing script was provided, if so, use it first
+    Map<String, String> env = System.getenv();
+    String preProccessingScript = env.getOrDefault("PREPROCESSING_SCRIPT_PATH", "");
+    
+    if (!preProccessingScript.isEmpty()) {
+      // Write request body to JSON file to be consumed by Python script
+      ObjectMapper mapper = new ObjectMapper();
+      
+      String fileName = UUID.randomUUID().toString();
+      
+      try {
+        mapper.writeValue(new File("/tmp/" + fileName), gcpRequest);
+        
+        // Run Python pre processing script and reload request body with updated data
+        ProcessBuilder processBuilder;
+        processBuilder = new ProcessBuilder("python", "/tmp/preprocessing_script.py", fileName);
```

Comment:
nit: I am not super comfortable with directly calling `python` here... given there is a possibility that future images have maybe different default python or something. I'd say no action necessary here but just for posterity I'm commenting.